lib/vmpooler/pool_manager.rb

```diff
@@ -26,6 +26,11 @@ def check_pending_vm(vm, pool, timeout, provider)
       end
     end
 
+    def slots_available?(slots, max_slots)
```

Comment:
Typically a method that ends in `?` is boolean.  Seems strange to return an object or nil.  Perhaps drop the question mark?